extensions/smallrye-jwt-build/deployment/src/main/java/io/quarkus/smallrye/jwt/build/deployment/SmallRyeJwtBuildProcessor.java

```diff
@@ -45,9 +47,15 @@ private void registerKeyLocationResource(Config config, String propertyName,
             BuildProducer<NativeImageResourceBuildItem> nativeImageResource) {
         Optional<String> keyLocation = config.getOptionalValue(propertyName, String.class);
         if (keyLocation.isPresent() && keyLocation.get().length() > 1
-                && (keyLocation.get().indexOf(':') < 0 || keyLocation.get().startsWith("classpath:"))) {
+                && (keyLocation.get().indexOf(':') < 0 || (keyLocation.get().startsWith(CLASSPATH_SCHEME)
+                        && keyLocation.get().length() > CLASSPATH_SCHEME_LENGTH))) {
             log.infof("Adding %s to native image", keyLocation.get());
             String location = keyLocation.get().startsWith("/") ? keyLocation.get().substring(1) : keyLocation.get();
+
+            // It can only be `classpath:` at this point
+            if (location.charAt(CLASSPATH_SCHEME_LENGTH - 1) == ':') {
```

Comment:
It's nitpicking but I think I would rather test for `.startsWith(CLASSPATH_SCHEME)` rather than being smart. It's a lot more clear, you don't even need a comment and you don't have to check the logic.

Also, it's not a hot path so there's no need to overoptimize.

Same for the other location.

Feel free to ignore if you have a strong opinion about it, though.

Reply:
@gsmet Sure, it makes sense